### DISCUSSION TITLE
native cat9-style shell

### DISCUSSION THREAD

**Discussion by christianparpart** [charmbracelet/inspo#40] [Ideas]
**Is your feature request related to a problem? Please describe.**

The README of cat9 isn't the clearest to me at least, TL;DR is that it acts a little bit like a TUI-style shell where you can run commands, just like a standard shell, except that the command output can be fold/unfold and you do not need to wait for a command to finish but already start the next. I really like the idea. So I looked up how to install that, boom. It's a plugin inside of another thing and the way you start it is certainly not what I'd call convenient.

**Describe the solution you'd like**

The project idea I'm having is actually what you can call **a asynchronous TUI command multiplexer**. A modernized version of cat9, but as a single binary, easily downloadable and runnable right out of the box with good defaults by default (e.g. visually appealing to the modern-ish terminal looking eye).

**Include projects that are similar in nature**

https://github.com/letoram/cat9

**Complexity**

That might sound like a fun project. True. But I think it's actually not what one should call trivial due to the asynchronously of running commands and most importantly, each command can receive input, so should be possible to switch focus and especially the command output needs to be VT parsed and support some kind of VT sequence re-interpretation, which -in fact- puts it right into the niche projects of being a terminal multiplexer (think of tmux/screen/zellij.

And implementation of such a tool could either go ahead and implement its own shell prompt or optionally additionally even go with existing shells by utilizing their shell integrations in order to find out where a command prompt starts and ends. However the latter induces another issue as you cannot run the output in the background with a single shell, requiring you to spawn a new shell for each new command. until completed.

**Comment by bashbunni** [comment]
Hey I'm just converting all of the project ideas in this repo to discussions instead of issues :)
